cpp/src/IceDiscovery/LookupI.cpp

```diff
@@ -212,50 +203,45 @@ LookupI::destroy()
 }
 
 void
-LookupI::setLookupReply(const LookupReplyPrxPtr& lookupReply)
+LookupI::setLookupReply(const LookupReplyPrx& lookupReply)
 {
-    //
-    // Use a lookup reply proxy whose adress matches the interface used to send multicast datagrams.
-    //
-    for(vector<pair<LookupPrxPtr, LookupReplyPrxPtr> >::iterator p = _lookups.begin(); p != _lookups.end(); ++p)
+    // Create one lookup proxy per endpoint from the given proxy. We want to send a multicast datagram on each
```

Comment:
I don't think we ever call `setLookupReply` more than once but should we clear `_lookups`  here just in case?Otherwise we'll keep pushing new entries into `_lookups`.

Reply:
Maybe just assert is empty?